CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+cmake_minimum_required(VERSION 3.16)
+project(oscprob LANGUAGES CXX)
+
+option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
+
+include(GNUInstallDirs)
+
+# Instructions to properly handle RPATH
+if(NOT CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+  set(CMAKE_RUNTIME_OUTPUT_DIRECTORY
+      ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR})
+endif()
+if(NOT CMAKE_LIBRARY_OUTPUT_DIRECTORY)
+  set(CMAKE_LIBRARY_OUTPUT_DIRECTORY
+      ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR})
+endif()
+if(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
+  set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY
+      ${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR})
+endif()
+
+#
+# Define our RPATH settings, for both build and install phases
+#
+
+# use, i.e. do not skip, the full RPATH in the _build_ tree
+set(CMAKE_SKIP_BUILD_RPATH FALSE)
+
+# when building, do not use the install RPATH already (will only be used when
+# actually installing)
+set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+
+# add to the install RPATH the (automatically determined) parts of the RPATH
+# that point to directories outside the build tree
+set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
+# specify libraries directory relative to binaries one.
+file(RELATIVE_PATH relDir ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}
+     ${CMAKE_LIBRARY_OUTPUT_DIRECTORY})
+
+if(APPLE)
+  set(basePoint @loader_path)
+else()
+  set(basePoint $ORIGIN)
+endif()
+
+set(CMAKE_INSTALL_RPATH ${basePoint} ${basePoint}/${relDir}
+                        ${basePoint}/../../lib)
+
+include(dependencies.cmake)
+
+# useful functions to ease declaration of libraries and executables for this
+# project
+include(cmake/functions/CMakeLists.txt)
+
+# Define SOURCES
+set(SOURCES
+    src/Absorption.cxx
+    src/EarthModelBase.cxx
+    src/EarthModelBinned.cxx
+    src/EigenPoint.cxx
+    src/NuPath.cxx
+    src/PMNS_Base.cxx
+    src/PMNS_Decay.cxx
+    src/PMNS_Deco.cxx
+    src/PMNS_DensityMatrix.cxx
+    src/PMNS_Fast.cxx
+    src/PMNS_Iter.cxx
+    src/PMNS_LIV.cxx
+    src/PMNS_NSI.cxx
+    src/PMNS_NUNM.cxx
+    src/PMNS_OQS.cxx
+    src/PMNS_SNSI.cxx
+    src/PMNS_Sterile.cxx
+    src/PremModel.cxx
+    src/complexsolver.cxx)
+
+# Define headers from sources
+list(TRANSFORM SOURCES REPLACE "src/" "inc/" OUTPUT_VARIABLE HEADERS)
+list(TRANSFORM HEADERS REPLACE ".cxx" ".h")
+list(APPEND HEADERS "inc/Definitions.h" "${CMAKE_BINARY_DIR}/prem_default.hpp")
+
+# Add library
+add_library(OscProb ${SOURCES})
+# This is needed if we want to compile this project from a dependency with
+# fetchContent
+add_library(OscProb::OscProb ALIAS OscProb)
+
+# Add include directories
+list(APPEND includePath ${CMAKE_CURRENT_SOURCE_DIR}
+     ${CMAKE_CURRENT_SOURCE_DIR}/inc ${CMAKE_BINARY_DIR})
+foreach(incDir IN LISTS includePath)
```

Comment:
Please note that a more "modern" (and/or future proof ?) approach for include dirs and header files might be to use [file sets](https://cmake.org/cmake/help/latest/command/target_sources.html#file-sets) (available for cmake >= 3.23). 

see a (random)  example in https://git.km3net.de/laphecetche/waveforms/-/blob/main/src/CMakeLists.txt?ref_type=heads#L10

Reply:
Done in the new version (but I force-pushed since I also needed to modify the name of the namespace in exceptions.h)